sdk/core/Azure.Core/tests/TransportFunctionalTests.cs

```diff
@@ -1079,9 +1085,18 @@ public async Task ServerCertificateCustomValidationCallbackIsHonored([Values(tru
         [Test]
         public async Task ClientCertificateIsHonored([Values(true, false)] bool setClientCertificate)
         {
+#if NETFRAMEWORK // ServicePointManager is obsolete and doesn't affect HttpClient
             // This test assumes ServicePointManager.ServerCertificateValidationCallback will be unset.
             ServicePointManager.ServerCertificateValidationCallback = null;
-            var clientCert = new X509Certificate2(Convert.FromBase64String(Pfx));
+#endif
+            byte[] cer = Convert.FromBase64String(Pfx);
+            X509Certificate2 clientCert;
+
+#if NET9_0_OR_GREATER
```

Comment:
[SYSLIB0057: X509Certificate2 and X509Certificate constructors for binary and file content are obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0057) - load certificates using APIs that depend on the intended content type, instead of using the obsoleted APIs that allow multiple formats.